cpp/include/raft/core/device_mdspan.hpp

```diff
@@ -266,23 +266,42 @@ auto make_device_matrix_view(ElementType* ptr, IndexType n_rows, IndexType n_col
  * @tparam LayoutPolicy policy for strides and layout ordering
  * @param[in] ptr on device to wrap
  * @param[in] n number of elements in pointer
- * @param[in] stride the stride between consecutive elements in the vector. Setting to a value
- *            other than 1 requires LayoutPolicy to be set to layout_stride
  * @return raft::device_vector_view
  */
-template <typename ElementType,
-          typename IndexType    = std::uint32_t,
-          typename LayoutPolicy = layout_c_contiguous>
-auto make_device_vector_view(ElementType* ptr, IndexType n, IndexType stride = 1)
+template <typename ElementType, typename IndexType, typename LayoutPolicy = layout_c_contiguous>
+auto make_device_vector_view(ElementType* ptr, IndexType n)
+{
+  return device_vector_view<ElementType, IndexType, LayoutPolicy>{ptr, n};
+}
+
+/**
+ * @brief Create a 1-dim mdspan instance for device pointer.
+ * @tparam ElementType the data type of the vector elements
+ * @tparam IndexType the index type of the extents
+ * @tparam LayoutPolicy policy for strides and layout ordering
+ * @param[in] ptr on device to wrap
+ * @param[in] mapping The layout mapping to use for this vector
+ * @return raft::device_vector_view
+ */
+template <typename ElementType, typename IndexType, typename LayoutPolicy = layout_c_contiguous>
+auto make_device_vector_view(
+  ElementType* ptr,
+  const typename LayoutPolicy::template mapping<vector_extent<IndexType>>& mapping)
+{
+  return device_vector_view<ElementType, IndexType, LayoutPolicy>{ptr, mapping};
+}
+
+/**
+ * @brief Construct a strided vector layout mapping
+ * @tparam IndexType the index type of the extents
+ * @params[in] n the number of elements in the vector
+ * @params[in] stride the stride between elements in the vector
+ */
+template <typename IndexType>
+auto make_vector_strided_layout(IndexType n, IndexType stride)
```

Comment:
I think this can be more reusable than just 1d. Of course, I don't mind having a specialized function for vectors, but perhaps we could add a more generalized function  like `make_strided_layout(extents, strides)` and then have `make_vector_strided_layout()` invoke `make_strided_layout`? Just thinking about reuse here. We can have strided layouts that aren't necessarily c- or f-contiguous.